forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/as/api/registration/dynamic/DynamicRegistrationApiController.java

```diff
@@ -271,34 +280,37 @@
                     String ssaSerialised = registrationRequestJws.getJWTClaimsSet()
                             .getStringClaim(RegistrationTppRequestClaims.SOFTWARE_STATEMENT);
 
+                    if(ssaSerialised == null){
+                        log.debug("No SSA provided in registationJWT");
+                        throw  new OBErrorException(OBRIErrorType.TPP_REGISTRATION_SSA_INVALID);
+                    }
+
                     //Convert in json for convenience
                     String registrationRequestJson =
                             JSONObjectUtils.toJSONString(registrationRequestJws.getJWTClaimsSet().toJSONObject());
                     OIDCRegistrationRequest oidcRegistrationRequest = objectMapper.readValue(
                             registrationRequestJson, OIDCRegistrationRequest.class);
 
-                    String directoryId;
-                    if (ssaSerialised == null && currentUser.getAuthorities().contains(OBRIRole.ROLE_EIDAS)) {
-                        ssaSerialised = generateSSAForEIDAS(authentication, registrationRequestJws, oidcRegistrationRequest);
-                        directoryId = ORIGIN_ID_EIDAS;
+                    String directoryId = tppRegistrationService.verifySSA(ssaSerialised);
+                    if (directoryId == null) {
+                        log.debug("None of the directories signed this SSA {}", ssaSerialised);
```

Comment:
I'm not sure if I've understood this correctly. Do directories sign SSAs, or are SSAs in the directories either signed or unsigned?